jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/PluginConfigurationProcessorTest.java

```diff
@@ -268,22 +273,78 @@ public void testPluginConfigurationProcessor_cacheDirectorySystemProperties()
     verify(containerizer).setApplicationLayersCache(Paths.get("/new/application/cache"));
   }
 
+  @Test
+  public void testAddJvmArgFilesLayer() throws IOException, InvalidAppRootException {
+    String classpath = "/extra:/app/classes:/app/libs/dep.jar";
+    String mainClass = "com.example.Main";
+    PluginConfigurationProcessor.addJvmArgFilesLayer(
+        rawConfiguration, projectProperties, jibContainerBuilder, classpath, mainClass);
+
+    Path classpathFile = appCacheDirectory.resolve("jib-classpath-file");
+    Path mainClassFile = appCacheDirectory.resolve("jib-main-class-file");
+    String classpathRead = new String(Files.readAllBytes(classpathFile), StandardCharsets.UTF_8);
+    String mainClassRead = new String(Files.readAllBytes(mainClassFile), StandardCharsets.UTF_8);
+    assertThat(classpathRead).isEqualTo(classpath);
+    assertThat(mainClassRead).isEqualTo(mainClass);
+
+    List<FileEntry> layerEntries =
+        getLayerEntries(jibContainerBuilder.toContainerBuildPlan(), "jvm arg files");
+    assertThat(layerEntries)
+        .comparingElementsUsing(SOURCE_FILE_OF)
+        .containsExactly(
+            appCacheDirectory.resolve("jib-classpath-file"),
+            appCacheDirectory.resolve("jib-main-class-file"));
+    assertThat(layerEntries)
+        .comparingElementsUsing(EXTRACTION_PATH_OF)
+        .containsExactly("/app/jib-classpath-file", "/app/jib-main-class-file");
+  }
+
+  @Test
+  public void testWriteFileConservatively() throws IOException {
```

Comment:
we may not need this test, given `testWriteFileConservatively_updatedContent()` below?

Reply:
My bad, looks good.